cmd/plugin/secret_generation.go

```diff
@@ -0,0 +1,454 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const kubectl = "kubectl"
+
+var rootCmd = &cobra.Command{
+	Use:   "secret-generation",
+	RunE:  secretGeneration,
+	Short: "Create a kubeconfig secret for the pod lister controller",
+}
+
+var (
+	contextName    string
+	name           string
+	namespace      string
+	serviceAccount string
+	dirty          bool = false
+)
+
+type Secret struct {
+	APIVersion string            `yaml:"apiVersion"`
+	Kind       string            `yaml:"kind"`
+	Metadata   Metadata          `yaml:"metadata"`
+	StringData map[string]string `yaml:"stringData"`
+}
+
+type Metadata struct {
+	Name      string            `yaml:"name"`
+	Namespace string            `yaml:"namespace"`
+	Labels    map[string]string `yaml:"labels"`
+}
+
+// Config represents a Kubernetes configuration file.
+type Config struct {
+	APIVersion     string         `yaml:"apiVersion"`
+	Kind           string         `yaml:"kind"`
+	Clusters       []NamedCluster `yaml:"clusters"`
+	Contexts       []NamedContext `yaml:"contexts"`
+	CurrentContext string         `yaml:"current-context"`
+	Users          []NamedUser    `yaml:"users"`
+}
+
+// NamedCluster holds the name and cluster details.
+type NamedCluster struct {
+	Name    string  `yaml:"name"`
+	Cluster Cluster `yaml:"cluster"`
+}
+
+// Cluster holds the server and certificate authority data.
+type Cluster struct {
+	Server                   string `yaml:"server"`
+	CertificateAuthorityData string `yaml:"certificate-authority-data"`
+}
+
+// NamedContext holds the name and context details.
+type NamedContext struct {
+	Name    string  `yaml:"name"`
+	Context Context `yaml:"context"`
+}
+
+// Context holds the cluster and user names.
+type Context struct {
+	Cluster string `yaml:"cluster"`
+	User    string `yaml:"user"`
+}
+
+// NamedUser holds the name and user details.
+type NamedUser struct {
+	Name string `yaml:"name"`
+	User User   `yaml:"user"`
+}
+
+// User holds the user's token.
+type User struct {
+	Token string `yaml:"token"`
+}
+
+func secretGenerationCommand() *cobra.Command {
+	return rootCmd
+}
+
+func init() {
+	rootCmd.Flags().StringVarP(&contextName, "context", "c", "", "kubeconfig context to use (required)")
+	if err := rootCmd.MarkFlagRequired("context"); err != nil {
+		panic(err)
+	}
+
+	rootCmd.Flags().StringVar(&name, "name", "", "name for the secret (defaults to context name)")
```

Comment:
Does it still default to context name, looks like it defaults to ""?

Reply:
Yes, on line 117 if the name has a length of zero it gets set to be the context name.